src/content/blog/astro-4120.mdx

```diff
@@ -0,0 +1,152 @@
+
+[server-islands.com](https://server-islands.com) performs consistently faster in repeated runs; it scores a 20% faster Largest Contentful Paint, which is one of Google's 3 Core Web Vitals impacting both user experience and SEO.
+
+Instead of starting the request to the origin on the server like PPR, Server Islands do so from the browser. This means that while Server islands paint earlier, they might finish later than PPR.
+
+Both Server islands and PPR work best in sites like these, where the main page content can be edge cached. They work well for ecommerce but would probably not be something you want to use for a social network feed. Due to the page being edge cached, that wait for the island request is minimal.
+
+Ultimately, we are confident in this tradeoff as being the best overall performance for Astro users across both the user experience and passing Core Web Vitals without requiring any special backend infrastructure or support to get started.
+
+### Portability
+
+We built **Server islands** with portability in mind. It doesn't depend on any server infrastructure so it will work with any host you have, from a Node.js server in a Docker container to the serverless provider of your choice.
+
+Our implementation happens mostly at build-time, where component content is swapped out for a small script, one that can almost fit into a Tweet.
+
+Each of the islands marked with `server:defer` are split off into their own special route which the script fetches at run time. The trade-off is that the island doesn't get a head-start rendering as it might if it depended on edge handling, but we think the performance numbers favor the simpler approach we've taken thus far.
+
+### Trying out Server Islands
+
+We're still working on polishing the feature, but you can try out Astro Server Islands today by enabling the feature in Astro 4.12:
+
+```js
+import { defineConfig } from 'astro/config';
+import netlify from '@astrojs/netlify';
+
+export default defineConfig({
+  output: 'hybrid',
+  adapter: netlify(),
+
+  experimental: {
+    serverIslands: true,
+  },
+})
+```
+
+Server islands, as the name implies, run on the server, so you'll need to use either `'server'` or `'hybrid'` output. Once you've enabled the feature you can use them on any component using the `server:defer` directive:
+
+```astro
+<Avatar server:defer>
+  <GenericUserImage slot="fallback" />
+</Avatar>
+```
+
+When Astro builds your site it will omit the component and inject a script in its place, in addition to the content you've marked with `slot="fallback"`. When the page loads in the browser these components will be requested, along with the props you passed to them, to a special endpoint that renders them and returns the HTML.
```

Comment:
```suggestion
When Astro builds your site it will omit the component and inject a script in its place, in addition to the content you've marked with `slot="fallback"`. When the page loads in the browser, these components along with any props you pass to them, will be requested to a special endpoint that renders them and returns the HTML.
```